Cesium3DTilesSelection/src/TilesetJsonLoader.cpp

```diff
@@ -0,0 +1,855 @@
+#include "TilesetJsonLoader.h"
+
+#include "ImplicitOctreeLoader.h"
+#include "ImplicitQuadtreeLoader.h"
+
+#include <Cesium3DTilesSelection/GltfConverters.h>
+#include <Cesium3DTilesSelection/TileID.h>
+#include <CesiumAsync/AsyncSystem.h>
+#include <CesiumAsync/IAssetResponse.h>
+#include <CesiumGeometry/BoundingSphere.h>
+#include <CesiumGeometry/OrientedBoundingBox.h>
+#include <CesiumGeospatial/BoundingRegion.h>
+#include <CesiumGeospatial/S2CellBoundingVolume.h>
+#include <CesiumUtility/JsonHelpers.h>
+#include <CesiumUtility/Uri.h>
+#include <CesiumUtility/joinToString.h>
+
+#include <rapidjson/document.h>
+#include <spdlog/logger.h>
+
+#include <cctype>
+
+namespace Cesium3DTilesSelection {
+namespace {
+struct ExternalContentInitializer {
+  // Have to use shared_ptr here to make this functor copyable. Otherwise,
+  // std::function won't work with move-only type as it's a type-erasured
+  // container. Unfortunately, std::move_only_function is scheduled for C++23.
+  std::shared_ptr<TilesetContentLoaderResult> pExternalTilesetLoaders;
+  TilesetJsonLoader* tilesetJsonLoader;
+
+  void operator()(Tile& tile) {
+    TileContent& content = tile.getContent();
+    if (content.isExternalContent()) {
+      std::unique_ptr<Tile>& pExternalRoot = pExternalTilesetLoaders->pRootTile;
+      if (pExternalRoot) {
+        // propagate all the external tiles to be the children of this tile
+        std::vector<Tile> children;
+        children.emplace_back(std::move(*pExternalRoot));
+        tile.createChildTiles(std::move(children));
+
+        // save the loader of the external tileset in this loader
+        tilesetJsonLoader->addChildLoader(
+            std::move(pExternalTilesetLoaders->pLoader));
+      }
+    }
+  }
+};
+
+void logErrors(
+    const std::shared_ptr<spdlog::logger>& pLogger,
+    const std::string& url,
+    const std::vector<std::string>& errors) {
+  if (!errors.empty()) {
+    SPDLOG_LOGGER_ERROR(
+        pLogger,
+        "Failed to load {}:\n- {}",
+        url,
+        CesiumUtility::joinToString(errors, "\n- "));
+  }
+}
+
+void logWarnings(
+    const std::shared_ptr<spdlog::logger>& pLogger,
+    const std::string& url,
+    const std::vector<std::string>& warnings) {
+  if (!warnings.empty()) {
+    SPDLOG_LOGGER_WARN(
+        pLogger,
+        "Warning when loading {}:\n- {}",
+        url,
+        CesiumUtility::joinToString(warnings, "\n- "));
+  }
+}
+
+void logErrorsAndWarnings(
```

Comment:
This function and the two above are repeated a few times, probably worth factoring out.

Reply:
This is now separated into a helper function into a different file in a unit test PR. I leave this comment here to mark it as temporarily resolved until the PR is merged